src/main/scala/diplomaticobjectmodel/logicaltree/LogicalTreeNode.scala

```diff
@@ -21,17 +31,43 @@ object LogicalModuleTree {
     tree.put(parent, treeNode)
   }
 
-  def root: LogicalTreeNode = {
+  def rootLogicalTreeNode: LogicalTreeNode = {
     val roots = tree.collect { case (k, _) if !tree.exists(_._2.contains(k)) => k }
     assert(roots.size == 1, "Logical Tree contains more than one root.")
     roots.head
   }
 
-  def bind(resourceBindingsMap: ResourceBindingsMap): Seq[OMComponent] = {
+  def getResourceBindings(device: Device, maps: ArrayBuffer[ResourceBindingsMap]): ResourceBindings = {
+    val rbm = maps.find {
+      rbm => rbm.map.contains(device)
+    }.getOrElse {
+      bindingScopes.foreach { s =>
+        val stuff = s.getResourceBindingsMap.map.keys.collect { case x: SimpleDevice => x }
+        println(s"BS: ${stuff.map(_.devname)}")
+      }
+      println(s"Device = ${device.asInstanceOf[SimpleDevice].devname} ")
```

Comment:
Should this debug code be committed?

Reply:
I want to leave it in until all of the BindingScope changed are done.